internal/util/neutrontags/join.go

```diff
@@ -28,8 +28,12 @@ import (
 	"github.com/k-orc/openstack-resource-controller/internal/osclients"
 )
 
+type StringTag interface {
+	orcv1alpha1.NeutronTag | orcv1alpha1.ServerTag | orcv1alpha1.ImageTag
+}
+
 // Join joins a slice of tags into a comma separated list of tags.
-func Join(tags []orcv1alpha1.NeutronTag) string {
+func Join[T StringTag](tags []T) string {
```

Comment:
We might want to rename this package, or more the function somewhere else.

Reply:
It was called NeutronTag because the functionality has a common implementation in Neutron. My only worry would be that Nova and Neutron could technically diverge on this. It's probably fine, though.